<?xml version="1.0"?>
<!-- ROS head file to simulate vizzy in gazebo -->
<!-- Now with the ability to perform gazing in the real one -->
<launch>
  <arg name="config_file" default="$(find vizzy_control)/config/frames.yaml" />
  <arg name="simulation" default="true" />
  <arg name="fixation_point_in" default="fixation_point" /> 
  <arg name="fixation_point_out" default="fixation_point_goal_ros"/>

  <arg name="joint_states" default="/vizzy/joint_states" /> 

  <group if="$(arg simulation)">
    <node name="gaze" pkg="vizzy_control" type="gaze_sim" respawn="true" output="screen">
      <rosparam command="load" file="$(find vizzy_moveit_config)/config/kinematics.yaml"/>
      <rosparam command="load" file="$(arg config_file)"/>
      <remap from="joint_states" to="$(arg joint_states)"/>
    </node>

    <node name="gaze_status" pkg="vizzy_control" type="fixation_point_node" respawn="true" output="screen" >
      <rosparam file="$(arg config_file)" command="load" />
    </node>
  </group>

  <group unless="$(arg simulation)">
    <node name="gaze" pkg="vizzy_control" type="gaze_real" respawn="true" output="screen">
      <rosparam command="load" file="$(find vizzy_moveit_config)/config/kinematics.yaml"/>
      <rosparam command="load" file="$(arg config_file)"/>
      <remap from="fixation_point_in" to="$(arg fixation_point_in)"/>
      <remap from="fixation_point_out" to="$(arg fixation_point_out)"/>
    </node>    
  </group>
</launch>
